oracle/price-feeder/oracle/oracle.go

```diff
@@ -116,19 +119,24 @@ func (o *Oracle) Start(ctx context.Context) error {
 		default:
 			o.logger.Debug().Msg("starting oracle tick")
 
-			startTime := time.Now()
+			// Wait for next block height to be available in the channel
+			nextBlockHeight := <-o.oracleClient.BlockHeightEvents
 
-			if err := o.tick(ctx); err != nil {
-				telemetry.IncrCounter(1, "failure", "tick")
-				o.logger.Err(err).Msg("oracle tick failed")
-			}
+			go func() {
+				startTime := time.Now()
+
+				if err := o.tick(ctx, clientCtx, txFactory, nextBlockHeight); err != nil {
+					telemetry.IncrCounter(1, "failure", "tick")
+					o.logger.Err(err).Msg(fmt.Sprintf("oracle tick failed for height %d", nextBlockHeight))
+				}
 
-			o.lastPriceSyncTS = time.Now()
+				o.lastPriceSyncTS = time.Now()
 
-			telemetry.MeasureSince(startTime, "runtime", "tick")
-			telemetry.IncrCounter(1, "new", "tick")
+				telemetry.MeasureSince(startTime, "runtime", "tick")
+				telemetry.IncrCounter(1, "new", "tick")
+
```

Comment:
can we also add metric for tick for chain height? then we can visualize missed ones  / eventualy alert on them better too

Reply:
what do you mean by missed ones?

Reply:
Do you mean catch the event subscribe missing to push some block height to the channel?

Reply:
i mean if the tick doesn't happen for a specific height because we had a newer height

Reply:
to go along with the comment for not processing older tick heights if a newer one is present

Reply:
I think this isnt actionable right now considering that theres no real expectation of backlog in the height channel